File: Leetcode/1. Two Pointers/Exercício 2 - Valid Palindrome.py
frase = "A man a plan a canal Panama"
# true

def isPalindrome(s):
    left, right = 0, len(s) - 1

    while left < right:
        if s[left] == s[right]:
            return True

        left += 1
        right -= 1

    return False


print(isPalindrome(frase))

File: Leetcode/1. Two Pointers/Exercício 2 - Valid Palindrome.js
frase = "A man a plan a canal Panama";
frase1 = "lorem canal Panama";
// True

function isValidPalidrome(frase) {
  let left = 0;
  let right = frase.length - 1;

  while (left < right) {
    if (frase[left] == frase[right]) return true;

    left++;
    right--;
  }

  return false;
}

console.log(isValidPalidrome(frase));
console.log(isValidPalidrome(frase1));
